### PR TITLE
docs: Transition md files to support new rtd format

### PR DESCRIPTION
**Pull request by NoahGorny**
After #1688

### CODE REVIEW

**Comment by NoahGorny** [comment]
@davidpfarrell take a look, this is yet another step, in here we will officially move into readthedocs!

**Comment by davidpfarrell** [comment]
Hi Team,

As much as I think there's value in using ReadTheDocs, I feel sad about losing the local `CONTRIBUTING.md` and essentially turning `README.md` into a shell of links to RTD.

I Also think there's value in having both of these files fleshed out and present *in* the repo.

I've done some light searching and haven't yet found any "best practices" for RTD integration in this regard, but ...

It was my thought that, by managing the documentation in the docs/*.rst files, we would be able to *both* allow our full documentation to be hosted on RTD *and* be able to render local files like these that live in the repo.

[INTERRUPTION]

So while looking for info on how to manage README's with RTD, I came across this tutorial:
* [How to Set Up Your Python Project Docs for Success](https://towardsdatascience.com/how-to-set-up-your-python-project-docs-for-success-aab613f79626)

The page contains this quote which resonates with me:
> Your README should include install and basic use information. I suggest you point users to the full docs on RTD.

Here also I discovered that GitHub can serve your readme directly from the `docs/` folder:
* https://docs.github.com/en/free-pro-team@latest/github/creating-cloning-and-archiving-repositories/about-readmes

This is demonstrated by the sample project used in the tutorial:
* https://github.com/pybraries/pybraries
You'll notice the readme lives in `docs/` but is rendered on the main page.

He includes the readme on the index page, but I could go either way on that idea.

So this seems like a decent compromise, ie
* Manage a useful README within the docs/ folder.

We could have some shared sections between the README and INDEX files that are managed by shared/included snippets, assuming Sphinx supports snippit-includes?

**Comment by NoahGorny** [comment]
> Hi Team,
> 
> As much as I think there's value in using ReadTheDocs, I feel sad about losing the local `CONTRIBUTING.md` and essentially turning `README.md` into a shell of links to RTD.
> 
> I Also think there's value in having both of these files fleshed out and present *in* the repo.
> 
> I've done some light searching and haven't yet found any "best practices" for RTD integration in this regard, but ...
> 
> It was my thought that, by managing the documentation in the docs/*.rst files, we would be able to *both* allow our full documentation to be hosted on RTD *and* be able to render local files like these that live in the repo.
> 
> [INTERRUPTION]
> 
> So while looking for info on how to manage README's with RTD, I came across this tutorial:
> * [How to Set Up Your Python Project Docs for Success](https://towardsdatascience.com/how-to-set-up-your-python-project-docs-for-success-aab613f79626)
> 
> The page contains this quote which resonates with me:
> > Your README should include install and basic use information. I suggest you point users to the full docs on RTD.
> 
> Here also I discovered that GitHub can serve your readme directly from the `docs/` folder:
> * https://docs.github.com/en/free-pro-team@latest/github/creating-cloning-and-archiving-repositories/about-readmes
> 
> This is demonstrated by the sample project used in the tutorial:
> * https://github.com/pybraries/pybraries
> You'll notice the readme lives in `docs/` but is rendered on the main page.
> 
> He includes the readme on the index page, but I could go either way on that idea.
> 
> So this seems like a decent compromise, ie
> * Manage a useful README within the docs/ folder.
> 
> We could have some shared sections between the README and INDEX files that are managed by shared/included snippets, assuming Sphinx supports snippit-includes?
> 
>  

Thank you @davidpfarrell for your thoughtful review. I understand your position about the `README` and `CONTRIBUTING` files, but I feel like the `README` is pretty bloated and unorganized. The `CONTRIBUTING` file can stay, but I do not think it's a must. We can merge docs with snippets as sphinx supports such methods. I am suggesting to wait for more reviews by @nwinkler and @cornfeedhobo

**Comment by davidpfarrell** [comment]
> I feel like the README is pretty bloated and unorganized.
Yeah I tend to agree, especially in light of the ability for RTD to document all the features.
I think the quote rings true:
> Your README should include install and basic use information.
I advocate for smaller, tighter README focused on introduction and installation, and directing to RTD for more, etc.
Whats great is that GitHub treads the docs/ folder as an official source of docs so we can manage the README there along with the rest of the docs.

> The CONTRIBUTING file can stay, but I do not think it's a must. 
In light of learning how GitHub treats the /docs folder, I think moving it to docs/contributing.rst is probably fine.
If the README could link to the repo version of the doc, I would find that better than linking to RTD version, but i'm not hung up on it.

> We can merge docs with snippets as sphinx supports such methods.
I thought I was seeing that feature by looking at the docs folder of the linked tutorial's example project, but wasn't sure - Good to know !

TL;DR: I'm okay with removing all .md files from root and managing them from docs/ ; Also think a minimal README should be present in docs/ since github will show it on repo main page.

**Comment by NoahGorny** [comment]
> > I feel like the README is pretty bloated and unorganized.
> Yeah I tend to agree, especially in light of the ability for RTD to document all the features.
> I think the quote rings true:
> > Your README should include install and basic use information.
> I advocate for smaller, tighter README focused on introduction and installation, and directing to RTD for more, etc.
> Whats great is that GitHub treads the docs/ folder as an official source of docs so we can manage the README there along with the rest of the docs.
> 
> > The CONTRIBUTING file can stay, but I do not think it's a must. 
> In light of learning how GitHub treats the /docs folder, I think moving it to docs/contributing.rst is probably fine.
> If the README could link to the repo version of the doc, I would find that better than linking to RTD version, but i'm not hung up on it.
> 
> > We can merge docs with snippets as sphinx supports such methods.
> I thought I was seeing that feature by looking at the docs folder of the linked tutorial's example project, but wasn't sure - Good to know !
> 
> TL;DR: I'm okay with removing all .md files from root and managing them from docs/ ; Also think a minimal README should be present in docs/ since github will show it on repo main page.
> 

Alright, I will move the README to the docs/ directory. Please note that contributing.rst is already converted and in the docs/ dir

**Comment by NoahGorny** [comment]
I think we can put some installation information in the new small README, and share the docs with the rtd using a plugin.. not sure if there are more things that should go there @davidpfarrell

File: docs/README.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
+# Bash-it
+
+[![Build Status](https://travis-ci.com/Bash-it/bash-it.svg?branch=master)](https://travis-ci.com/Bash-it/bash-it)
+[![Join the chat at https://gitter.im/Bash-it/bash-it](https://badges.gitter.im/Join%20Chat.svg)](https://gitter.im/Bash-it/bash-it?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge&utm_content=badge)
```
Comment (cornfeedhobo) [inline docs/README.md:4]: I realize not many people use irc, but could the Freenode channel be included as well?

https://webchat.freenode.net/?channel=#bash-it

Comment (NoahGorny) [reply docs/README.md:4]: This is a good idea, but I think it belongs to a separate PR

I will create an issue to address this :smile:

**Comment by NoahGorny** [comment]
@nwinkler @cornfeedhobo @davidpfarrell  what do you say guys? should we start the transition to 'rst'?

I have multiple PRs that are ready to be merged to the new docs, and I am ready to continue to improve the docs there (it will be easier to build and test, and more organized)

**Comment by NoahGorny** [comment]
Bumping @nwinkler @cornfeedhobo @davidpfarrell

**Comment by cornfeedhobo** [comment]
Sorry, I haven't commented because I just don't have a strong enough opinion. That said, I see no reason not to proceed.

**Comment by nwinkler** [comment]
Same - please go ahead!

**Comment by NoahGorny** [comment]
Alright- I will proceed to move the docs into `readthedocs`!  
I will create more PRs today to try to improve the docs even more
